python/GafferUI/CompoundEditor.py

```diff
@@ -998,3 +1040,126 @@ def __getSharedDragWidget( self ) :
 
 	# /nasty-fudge
 
+
+## Returns the preferred bound for a widget's window in NDC space
+# (relative to the available screen area). We use the relative area to
+# help with portability between platforms/window managers.
+# '-1' is used to indicate windows that are on the primary screen.
+# If a window is full screen, we still include the bound as otherwise
+# the resultant placement when the window is un-fullscreen'd can be a bit small.
+#
+# Note: We use bottom-left NDC coordinates to be consistent with Gaffer
+def _preferredBound( gafferWindow ) :
+
+	qWidget = gafferWindow._qtWidget()
+
+	window = qWidget.windowHandle()
+	if not window :
+		return {}
+
+	widgetScreen = window.screen()
+	widgetScreenNumber = QtWidgets.QApplication.desktop().screenNumber( qWidget )
+
+	if widgetScreen == QtWidgets.QApplication.primaryScreen():
+		widgetScreenNumber = -1
+
+	screenGeom = widgetScreen.availableGeometry()
+	screenW = float( screenGeom.width() )
+	screenH = float( screenGeom.height() )
+
+	# Ideally we'd use frameGeometry for portability, but have seen a variety
+	# of issues where the window has been made visible, but doesn't yet
+	# have a frame, as we rely on asking for the margins (as you can't
+	# setFrameGeometry) we end up with unreliable window placement. As such
+	# we compromise and store the widgets geometry instead.
+	windowGeom = qWidget.normalGeometry()
+
+	x = ( windowGeom.x() - screenGeom.x() ) / screenW
+	y = 1.0 - ( ( windowGeom.y() - screenGeom.y() ) / screenH )
+	w = windowGeom.width() / screenW
+	h = windowGeom.height() / screenH
+
+	return {
+		"screen" :  widgetScreenNumber,
+		"fullScreen" : gafferWindow.getFullScreen(),
+		"maximized" : bool(window.windowState() & QtCore.Qt.WindowMaximized),
+		# As we're bottom-left not top-left the y values are the other way around
+		"bound" : imath.Box2f( imath.V2f( x, y - h ), imath.V2f( x + w, y ) )
+	}
+
+def _imathRepr( data ) :
+
+	# Neither IECore.repr or repr include the module name
```

Comment:
Are you sure? I just tested and `IECore.repr()` does include the name. We rely on this all the time in other serialisations...

Reply:
Huh, seems to, definitely wasn't for a while when I was testing! Ill look into it some more

Reply:
```import imath
v = imath.V2f()
IECore.repr(v)
> imath.V2f( 0, 0 )
d = { "v" :v }
IECore.repr(d)
> {'v': V2f(0, 0)}
```

Reply:
Changed workaround in https://github.com/GafferHQ/gaffer/pull/3167/commits/4361ca224e657be1db373d6d0fe91d18a252c235